### PR TITLE
Moving prod deps to dev

### LINKED ISSUE #2531: Correct prod and dev dependencies in package.json
**Is your feature request related to a problem? Please describe.**
We are listing all deps we need to build the application in production deps and those we use only for dev in dev deps. This isn't quite right.

**Describe the solution you'd like**
The only dependencies that should be in the main dependency list are ones used at runtime. Therefore the dev dependencies aren't only ones used in development but also ones used during build stage.

**Describe alternatives you've considered**
n/a

**Testing notes**
Testing of all parts needed, probably worth a deploy test by pushing the branch to the test environment.

**Additional context**
Suggest the implementor checks best practices re non-obvious examples. For example, babel is a dev dependency but some of it's libraries may be used at runtime.

Also NB this almost certainly also requires our TravisCI and Jenkins files being updated.

### PR DESCRIPTION
**Pull request by thekp**
Resolves #2531

**Overall change:** _move build deps to devDep and update productionTest in jenkins to install all deps and uninstall once tests are over_

**Code changes:**

- _update package.json_
- _update make install to ensure node_modules are removed first_
- _update prodTest to install all dependencies_

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [x] Test engineer approval

### CODE REVIEW

**Comment by thekp** [comment]
This PR:(https://github.com/bbc/simorgh-infrastructure/pull/623) needs to be merged before this one.

File: package.json [hunk 1]
```diff
@@ -111,28 +111,22 @@
     "@bbc/psammead-styles": "^1.1.5",
     "@bbc/psammead-timestamp-container": "^2.1.0",
     "@bbc/psammead-visually-hidden-text": "^1.1.0",
-    "assets-webpack-plugin": "^3.9.8",
     "babel-loader": "^8.0.5",
```
Comment (dr3) [inline package.json]: Howcome were not moving babel?

Comment (thekp) [reply package.json]: 🤦‍♂

File: package.json [hunk 2]
```diff
@@ -149,15 +130,18 @@
     "speculate": "^2.0.0",
```
Comment (dr3) [inline package.json]: `speculate` can prob move too

**Review by dr3 — APPROVED** [review]

**Review by chris-hinds — APPROVED** [review]

**Comment by chris-hinds** [comment]
@thekp nice work, just needs the conflicts resolving.

**Comment by staylos92** [comment]
This doesn't need testing by a tester

**Pull request merged**